### ISSUE TITLE
typo in private param name: decodeDepreacted (probably should be decodeDeprecated)

### ISSUE THREAD

**Issue by RandalSchwartz** [flutter/flutter#125251]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Look at flutter/packages/flutter/lib/src/painting/image_provider.dart
2. Skip down to line 1092.
3. Chuckle at the typo

### Expected results

Proper spelling.  Not referenced anywhere but within the function as the local name for a positional parameter.

### Actual results

Chuckling.

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report @RandalSchwartz 
I looked at https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/painting/image_provider.dart and couldn't find the typo `decodeDepreacted`.  There are 9 instance of `decodeDeprecated` and all seem to be correctly typed, unless I missed to spot it :)

<img width="1204" alt="Screenshot 2023-04-21 at 10 55 49 AM" src="https://user-images.githubusercontent.com/67046386/233547735-d18f9acf-4fb2-4ec7-9404-8144abd04802.png">

Also, line 1092 is:

https://github.com/flutter/flutter/blob/d186792c00f61149916f685e2975300342f64749/packages/flutter/lib/src/painting/image_provider.dart#L1092

**Comment by RandalSchwartz** [comment]
commit 8d2978af642176f1e292edacc19d1a134fa3669f introduces it, but I don't see anything that removed it yet, and yet, it's not in the beta branch.  Weird.

But I'm sure I saw it in stable.

[more research]

Ahh, it was fixed in b319938ecd, which is now what's in beta.

**Comment by darshankawar** [comment]
Good to know. Closing based on above comment as fixed.
